### PR TITLE
Add CI labels

### PR DESCRIPTION
**Pull request by graceannobrien**
This PR addresses three issues:

1. Confidence intervals
- carries specified CI level through all functions 
- for example, if user selects 95 CI in `define_param_table`, CI column will then be named `ci_95` in `define_param_table` and `format_param_table` output. Then `make_pmtable` will then reformat and produce column `95\\% CI` to use for generating table(s).

2. Generate specific parameter tables by filtering and using pmtables
- after running `define_param_table` and `format_param_table`, user is ready to generate final parameter tables. 
- there are 5 generic parameter tables that `pmparams::make_pmtables` supports: full, fixed, fixed and random, fixed structural, fixed covariate, random. Defaults to "full".
- user has option to add more panels, notes. etc. to parameter table after `make_pmtable` is called

```
newDF <- define_param_table(.estimates = paramEst, .key = paramKey, .ci = 95, .zscore = NULL)
newFormatDF <- format_param_table(newDF)
pm_tibble1 <- make_pmtable(.df = newFormatDF, .pmtype = "full")
```

4. Vignettes
- `getting-started` and `making-bootstrap-parameter-tables` vignettes

### CODE REVIEW

File: R/make_pmtable.R [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+   dplyr::filter(stringr::str_detect(greek, "Omega") |
+                  stringr::str_detect(type, "Resid"))
+
+} else if (.pmtype == "fixed structural") {
+  pm_tab0 <- .df0 %>%
+    dplyr::filter(stringr::str_detect(type, "Struct"))
+
+} else if (.pmtype == "fixed covariate") {
+  pm_tab0 <- .df0 %>%
+    dplyr::filter(stringr::str_detect(type, "Covariate"))
```
Comment (graceannobrien) [inline R/make_pmtable.R:70]: Change to stringr::str_detect(type, "effect"))

File: R/make_pmtable.R [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+    pmtables::st_rename("Estimate" = "value")
+} else {
+  pm_tab1 <- pm_tab0 %>%
+    dplyr::select(-desc) %>%
+    pmtables::st_new() %>%
+    pmtables::st_panel("type") %>%
+    pmtables::st_blank("abb", "greek") %>%
+    pmtables::st_rename("Estimate" = "value",
+                        "Shrinkage (\\%)" = "shrinkage")
+}
```
Comment (graceannobrien) [inline R/make_pmtable.R:99]: `fixed and random`- keep description

File: R/make_pmtable.R [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+
+if (is.null(.notes)){
+  pm_tab2 <- pm_tab1
+
+} else {
+  pm_tab2 <- pm_tab1 %>%
+        pmtables::st_notes(.notes) %>%
+        pmtables::st_notes_detach(width = .width, type = "minipage")
```
Comment (graceannobrien) [inline R/make_pmtable.R:107]: remove "minipage"- not needed

**Comment by graceannobrien** [comment]
Currently failing drone because vignettes use `bbr`

**Comment by seth127** [comment]
@graceannobrien can you make the change that I proposed [here](https://github.com/metrumresearchgroup/pmparams/pull/32#pullrequestreview-2046317380) in this branch before @andersone1 reviews and we merge it? With @kyleam's correction, it would be changing [define_boot_table.R#L68](https://github.com/metrumresearchgroup/pmparams/blob/440cdf2a3681c4101860047f57409e7df85f13ad/R/define_boot_table.R#L68) to 

```r
if(all(c("parameter_names", "p50", "p2.5", "p97.5") %in% names(.bootParam0))) {
```

I looked into this more and realized that the updated `bbr` output actually has an additional column (`"original"`), so the way it's coded now, that `if()` block will actually _never trigger_.

```
packageVersion("bbr")
# [1] ‘1.10.0.8003’

param_estimates_compare(boot_sum, .orig_mod = orig_mod) 
# A tibble: 15 × 5                                                                                                                                                                 
   parameter_names original     p50    p2.5  p97.5
   <chr>              <dbl>   <dbl>   <dbl>  <dbl>
 1 THETA1            0.443   0.445   0.328  0.611 
 2 THETA2            4.12    4.12    4.06   4.17  
 3 THETA3            1.17    1.17    1.12   1.22  
 4 THETA4            4.21    4.21    4.17   4.24  
 5 THETA5            1.28    1.28    1.22   1.35  
 6 THETA6            0.485   0.479   0.403  0.548 
```

(As an aside, this is still currently in CI, because we aren't testing against the unreleased `bbr` yet, but it would start failing once that's released in a few weeks.)

**Comment by graceannobrien** [comment]
> @graceannobrien can you make the change that I proposed [here](https://github.com/metrumresearchgroup/pmparams/pull/32#pullrequestreview-2046317380) in this branch before @andersone1 reviews and we merge it? With @kyleam's correction, it would be changing [define_boot_table.R#L68](https://github.com/metrumresearchgroup/pmparams/blob/440cdf2a3681c4101860047f57409e7df85f13ad/R/define_boot_table.R#L68) to
> 
> ```r
> if(all(c("parameter_names", "p50", "p2.5", "p97.5") %in% names(.bootParam0))) {
> ```
> 
> I looked into this more and realized that the updated `bbr` output actually has an additional column (`"original"`), so the way it's coded now, that `if()` block will actually _never trigger_.
> 
> ```
> packageVersion("bbr")
> # [1] ‘1.10.0.8003’
> 
> param_estimates_compare(boot_sum, .orig_mod = orig_mod) 
> # A tibble: 15 × 5                                                                                                                                                                 
>    parameter_names original     p50    p2.5  p97.5
>    <chr>              <dbl>   <dbl>   <dbl>  <dbl>
>  1 THETA1            0.443   0.445   0.328  0.611 
>  2 THETA2            4.12    4.12    4.06   4.17  
>  3 THETA3            1.17    1.17    1.12   1.22  
>  4 THETA4            4.21    4.21    4.17   4.24  
>  5 THETA5            1.28    1.28    1.22   1.35  
>  6 THETA6            0.485   0.479   0.403  0.548 
> ```
> 
> (As an aside, this is still currently in CI, because we aren't testing against the unreleased `bbr` yet, but it would start failing once that's released in a few weeks.)

This is addressed in PR #34

**Pull request merged**